### PR TITLE
Make thermal contact work in 2D geometries not confined to or parallel to the x-y plane

### PR DESCRIPTION
**Pull request by gambka**
Refs #11913.

Had to regold the gap_heat_transfer_htonly_zr_test_out.e because the mesh file simple_2D.e was changed from containing SHELL4 to QUAD4 as is typically used in MOOSE.

### CODE REVIEW

**Comment by gambka** [comment]
@bwspenc @jasondhales

**Comment by gambka** [comment]
I like how the testing fails when you add new tests including requirements, design and issues but do not add them to existing tests in the tests directory for which you were not responsible for in the first place.

Is it now the responsibility of the next person to touch a tests file to add the requirements, design, and issues for all tests in the file? Seems like it should be the responsibility of the original contributors of those tests and not the next person to touch the file.

**Comment by permcody** [comment]
This isn't a perfect system, but it's also a problem that won't plague us for long.

**Comment by gambka** [comment]
If I have to update the tests files for the other tests. How do I go about finding what issue to assign to certain tests? Some of the tests were added back in the SVN days and don't have issues associated with them. Do I pick one of the later modifications to the tests that does have an issue? How does one decide what issues to assign to the test in the tests file?

**Comment by permcody** [comment]
We've been searching the history of the tests files to see what commits were made. Sometimes you can find something, even if it's somewhat modern and not the original issue that doesn't exist. Anything will do that might be useful for reconstructing the past. Do your best, if you need help finding something let me know!

**Comment by gambka** [comment]
While looking through the existing tests I've found that the ZR test added in #12071 does not provide the correct results.  A heat conduction only test should provide the same fluxes (and they should be conserved from one surface to the other) whether the axisymmetric axial axis is Y or X.  The RZ case  (with axial direction in Y) provides the correct results and has an analytical solution within the tests file.  The ZR case does not compute these same values as it should.

I have spoken with @vincentlaboure about the test.  I know what the requirement for the test would be but I am hesitant to put that the test 'shall' do something when it currently isn't doing it correctly.  

I will try taking a look at what is causing the discrepancy as part of this PR even if it is not directly related to this PR.

**Comment by gambka** [comment]
I've taken a look at the ZR test which provides incorrect results.  It appears to be much more involved (beyond the scope of this PR) to fix. I was planning to skip that test in the tests file and add the requirements, design, and issues for the other tests.  Are these parameters required if the test is skipped?

File: modules/combined/test/tests/gap_heat_transfer_htonly/tests [hunk 1]
```diff
@@ -1,67 +1,134 @@
...
+    design = 'source/bcs/GapHeatTransfer.md'
+    issues = '#1609'
   [../]
-
   [./3D_Iters]
     type = Exodiff
     input = 'gap_heat_transfer_htonly_it_plot_test.i'
     exodiff = 'out_it_plot.e'
     custom_cmp = 'gap_heat_transfer_htonly_it_plot_test.cmp'
+    requirement = "Thermal contact shall solve plate heat transfer for a constant conductivity gap in 3D at each iteration"
```
Comment (gambka) [inline modules/combined/test/tests/gap_heat_transfer_htonly/tests]: @permcody If I go back through the history you originally added this 3D_Iters test and I do not really know the difference between the 3D test above and this one other than in the Outputs block there is a parameter I have never seen or used before: `nonlinear_residual_dt_divisor`.  Perhaps you have a better idea of what the difference in requirements would be between this test and the one above it.

Comment (permcody) [reply modules/combined/test/tests/gap_heat_transfer_htonly/tests]: Acknowledging that I've seen this. I probably won't have a chance to look at it until Monday.

**Comment by gambka** [comment]
Another unrelated test failure.

**Comment by gambka** [comment]
How do we go about avoiding these unrelated timeout failures holding up PRs?

**Comment by permcody** [comment]
Short Answer: We all write better tests.

We are constantly watching this as they happen and refactoring tests. This particular test just got refactored last week. If you were to rebase, you'd get the newer version that won't time out anymore. Of course you don't have to do that, I'll just invalidate this and it'll likely pass given another chance. The problem here has already been fixed.

File: modules/combined/test/tests/gap_heat_transfer_htonly/tests (hunk 1 above)
Comment (gambka) [reply modules/combined/test/tests/gap_heat_transfer_htonly/tests]: Any chance to take a look at the test and verify whether the requirement I have listed is sufficient?

Comment (permcody) [reply modules/combined/test/tests/gap_heat_transfer_htonly/tests]: This is definitely not my test. It looks like I moved it way back in the module days but I can't seem to follow history before that. I might be @jasondhales test... 

At any rate, I think what you've done here is good and not worth holding up this PR any longer.

File: modules/heat_conduction/doc/content/source/bcs/GapHeatTransfer.md [hunk 2]
```diff
@@ -15,6 +15,8 @@ in the heat flux on the two surfaces.
 It is also important to use the appropriate `gap_geometry_type` parameter (PLATE, CYLINDER, or
 SPHERE) for the model geometry.
 
+Two-dimensional Cartesian geometries are not restricted to be in or parallel to the X-Y coordinate plane.
```
Comment (bwspenc) [inline modules/heat_conduction/doc/content/source/bcs/GapHeatTransfer.md]:
```suggestion
Two-dimensional Cartesian geometries are not restricted to be in or parallel to the X-Y coordinate plane. They can also lie in the X-Z or Y-Z coordinate planes.
```

**Review by bwspenc — APPROVED** [review]
Looks good.  One minor suggestion, but this is OK without it.

**Pull request merged**